compiler/optimizer/Inliner.cpp

```diff
@@ -1790,7 +1790,11 @@ TR_InlinerBase::addGuardForVirtual(
       {
       createdHCRAndVirtualGuard = true;
 
+      // we merge virtual guards and OSR guards for simplicity in most modes
+      // when using OSR to implement HCR we keep the HCR guards distinct since they
+      // will undergo special processing later in the compilation
       if (virtualGuard &&
+          comp()->getHCRMode() != TR::osr &&
```

Comment:
Do you think we could just check the HCR mode inside supportsMergingOfHCRGuards instead ? It would help with the other callers of that query in theory as well.